packages/core/src/components/date-picker/date-picker.tsx

```diff
@@ -8,73 +8,199 @@
  */
 
 import {
-    Component,
-    Event,
-    EventEmitter,
-    Fragment,
-    h,
-    Host,
-    Prop,
-    State
+  Component,
+  Event,
+  EventEmitter,
+  Fragment,
+  h,
+  Host,
+  Method,
+  Prop,
+  State,
 } from '@stencil/core';
 import { DateTime, Info, MonthNumbers } from 'luxon';
 import { DateTimeCardCorners } from '../date-time-card/date-time-card';
+import { DateChangeEvent, LagacyDateChangeEvent } from './events';
 
 @Component({
   tag: 'ix-date-picker',
   styleUrl: 'date-picker.scss',
   scoped: true,
 })
 export class DatePicker {
+  private daysInWeek = 7;
+  private dayNames = Info.weekdays();
+  private monthNames = Info.months();
+
   /**
-   * output date format
+   * Date format string.
+   * See {@link https://moment.github.io/luxon/#/formatting?id=table-of-tokens} for all available tokens.
    */
   @Prop() format: string = 'yyyy/LL/dd';
 
   /**
-   * Set range size
+   * If true a range of dates can be selected.
    */
   @Prop() range: boolean = true;
 
   /**
-   * set styles
+   * @deprecated - will get removed with next major release
    */
   @Prop() individual: boolean = true;
 
   /**
-   * Set corners style
+   * Corner style
    */
   @Prop() corners: DateTimeCardCorners = 'rounded';
 
-  private daysInWeek = 7;
-  private dayNames = Info.weekdays();
-  private monthNames = Info.months();
-  @State() year = DateTime.now().year;
-  @State() month = DateTime.now().month;
-  @State() calendar: [number, number[]][] = [];
+  /**
+   * Picker date. If the picker is in range mode this property is the start date.
+   *
+   * Format is based on `format`
+   *
+   * @since 1.1.0
+   */
+  @Prop() from: string = DateTime.now().toFormat(this.format);
+
+  /**
+   * Picker date. If the picker is in range mode this property is the end date.
+   * If the picker is not in range mode leave this value `null`
+   *
+   * Format is based on `format`
+   *
+   * @since 1.1.0
+   */
+  @Prop() to: string | null = null;
+
+  /**
+   * The earliest date that can be selected by the date picker.
+   * If not set there will be no restriction.
+   */
+  @Prop() minDate: DateTime;
+
+  /**
+   * The latest date that can be selected by the date picker.
+   * If not set there will be no restriction.
+   */
+  @Prop() maxDate: DateTime;
```

Comment:
Why is minDate and maxDate of type DateTime?
We should use string minDate and maxDate and use the format to calculate the DateTime internally. Otherwise we expose to much of our internal luxon logic

Reply:
Latest main is merged we must also add `@since 1.1.0` to new props/events/methods/slots